nntrainer/layers/layer_devel.h

```
@@ -251,6 +251,8 @@ class Layer {
    */
   virtual void setProperty(const std::vector<std::string> &values) = 0;
 
+  virtual std::string getProperty(const std::string &key) { return ""; }
```

It should raise the error in this function.

---

getProperty is not a function used by all layers.
Since it's only needed for specific layers, making it a pure virtual function would require all layers inheriting from layer_devel to implement it.

---

Yes, so what about raising error like NYI in this function since this body will be called only for layers not overriding this?